src/cloud/project/magento-app-php-application.md

```diff
@@ -41,96 +41,19 @@ php -m
 
 For details about a specific PHP extension, see the [PHP Extension List](https://www.php.net/manual/en/extensions.alphabetical.php).
 
-{{site.data.var.ece}} supports the following extensions:
+The following table show the supported PHP extensions when deploying {{site.data.var.ee}} on the Cloud platform.
 
--  Default extensions:
-   -  `bcmath`
-   -  `bz2`
-   -  `calendar`
-   -  `exif`
-   -  `gd`
-   -  `gettext`
-   -  `intl`
-   -  `mysqli`
-   -  `pcntl`
-   -  `pdo_mysql`
-   -  `soap`
-   -  `sockets`
-   -  `sysvmsg`
-   -  `sysvsem`
-   -  `sysvshm`
-   -  `opcache`
-   -  `zip`
+- **Default extensions** are the default extensions available with the PHP application.
+- **Required extensions** are installed automatically and cannot be removed
+- **Optional extensions**–You can install and remove these extensions as needed
 
--  Extensions that are installed and cannot be uninstalled:
-   -  `ctype`
-   -  `curl`
-   -  `date`
-   -  `dom`
-   -  `fileinfo`
-   -  `filter`
-   -  `ftp`
-   -  `hash`
-   -  `iconv`
-   -  `json`
-   -  `mbstring`
-   -  `mysqlnd`
-   -  `openssl`
-   -  `pcre`
-   -  `pdo`
-   -  `pdo_sqlite`
-   -  `phar`
-   -  `posix`
-   -  `readline`
-   -  `session`
-   -  `sqlite3`
-   -  `tokenizer`
-   -  `xml`
-   -  `xmlreader`
-   -  `xmlwriter`
-
--  Extensions that can be installed and uninstalled as needed:
-   -  `bcmath`
-   -  `bz2`
-   -  `calendar`
-   -  `exif`
-   -  `gd`
-   -  `geoip`
-   -  `gettext`
-   -  `gmp`
-   -  `igbinary`
-   -  `imagick`
-   -  `imap`
-   -  `intl`
-   -  `ioncube`
-   -  `ldap`
-   -  `mailparse`
-   -  `mcrypt`
-   -  `msgpack`
-   -  `mysqli`
-   -  `oauth`
-   -  `opcache`
-   -  `pdo_mysql`
-   -  `propro`
-   -  `pspell`
-   -  `raphf`
-   -  `recode`
-   -  `redis`
-   -  `shmop`
-   -  `soap`
-   -  `sockets`
-   -  `sodium`
-   -  `ssh2`
-   -  `sysvmsg`
-   -  `sysvsem`
-   -  `sysvshm`
-   -  `tidy`
-   -  `xdebug`
-   -  `xmlrpc`
-   -  `xsl`
-   -  `yaml`
-   -  `zip`
-   -  `pcntl`
+| Default extensions | Required extensions | Optional extensions |
```

Comment:
Required Extension list has been repeated in the Default Extensions list 
Also as the Optional list is much longer that the other, the DE & RE list float in the middle of the table. Can these be aligned to the top?

Reply:
Redid the table to begin with required extensions, followed by default and optional extensions.  I also removed the default extensions from the optional list, and updated the _Optional descriptions_  summary to state that the default extensions can be disabled.

Reply:
I'm afraid the list is still incorrect.
The following extensions are required but appear in the optional list -
bcmath
gd
intl
pdo_mysql
soap
xsl
zip

(https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html)

Reply:
The `sockets` extension also [became required in Magento 2.4.0](https://github.com/magento/magento2/pull/28585#issuecomment-643337713) and higher btw. Should also be included in https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html in my opinion!

Reply:
> I'm afraid the list is still incorrect.
> The following extensions are required but appear in the optional list -
> bcmath
> gd
> intl
> pdo_mysql
> soap
> xsl
> zip
> 
> (https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html)

These are listed in the default list in the original topic.  So I understood this as being installed by default, but you can remove them. 

How does one know they are required?

Reply:
so I see the issue:  On Magento Commerce, these extensions are required according to list in [Magento Commerce Install Guide](https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html).  

So on Cloud, are these extensions no longer required, or is the list of required PHP extensions for Cloud incorrect.

Reply:
@hostep  I updated the PHP settings topic to add a note about the ext-sockets extension that was in the 2.3 version of the topic:

```{:.bs-callout-warning}
If you install Magento via cloning from the [GitHub](https://github.com/magento/magento2) repository, then make sure you have the [ext-sockets](https://github.com/php-amqplib/php-amqplib/blob/master/CHANGELOG.md#281---2018-11-13) extension installed on your instance.
```
In the php-settings topic, the list of required extensions is generated in this file: https://github.com/magento/devdocs/blob/master/src/_includes/install/php-extensions-template.md, but `ext-sockets` is not included.  I think we should open a separate issue to address this, and request assistance from @dshevtsov.

Reply:
> I'm afraid the list is still incorrect.
> The following extensions are required but appear in the optional list -
> bcmath
> gd
> intl
> pdo_mysql
> soap
> xsl
> zip
> 
> (https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html)

I restored the original content from this topic.  I realized that I was conflating enable and disable extensions with install and uninstall.

Reply:
@meker12: we already kind-of had an issue for this, but it was closed unfortunately: https://github.com/magento/devdocs/issues/7568#issuecomment-694268314, should I create a new issue for this? I know colleagues of yours are [already aware of the issue for many months](https://github.com/magento/devdocs/pull/7398#issuecomment-644978773) (internal ticket DOC-80), but maybe we need an official public ticket to track this, otherwise it might be forgotten...

Reply:
@hostep Thanks for the context about the ext-sockets extension.  It's useful that the current guide now includes a specific mention of the ext-sockets extension, and why you need it.  Will follow up to understand the best information to provide about this extension.

Reply:
@hostep the DOC-80 is closed. It was mostly focused on the [Packages](https://devdocs.magento.com/guides/v2.4/release-notes/packages-open-source.html) topics. As it appeared, it's impossible to get accurate information about PHP versions from the Composer files. I hope that it is still possible for the list of extensions. Please open an issue for this with your recent suggestions (#7568, #7398) and assign it to me.

Reply:
@dshevtsov, done: https://github.com/magento/devdocs/issues/7960 (but I can't assign you)